pkg/devfile/image/docker_compatible.go

```diff
@@ -34,9 +35,22 @@ func (o *DockerCompatibleBackend) Build(image *devfile.ImageComponent, devfilePa
 	buildSpinner := log.SpinnerNoSpin("Building image locally")
 	defer buildSpinner.End(false)
 
-	shell := getShellCommand(o.name, image, devfilePath)
+	err := os.Setenv("PROJECTS_ROOT", devfilePath)
+	if err != nil {
+		return err
+	}
+
+	err = os.Setenv("PROJECT_SOURCE", devfilePath)
+	if err != nil {
+		return err
+	}
 
-	cmd := exec.Command("bash", "-c", shell)
+	//get command to run, expand the environment variables and convert command to array
+	shellCmd, err := util.ParseCommandLine(os.ExpandEnv(getShellCommand(o.name, image, devfilePath)))
```

Comment:
could we modify `getShellCommand` to return `[]string` so we can skip `ParseCommandLine`: